### PR TITLE
site-wide social include

### PR DESCRIPTION
**Pull request by aaronmarkham**
## Features
Social logos appear on the upper right. They move orientation as the screen gets smaller. Then disappear when it goes into mobile view. The hamburger menu now has a text version of the social options. I also threw the forum in that list.

There was some duplication and discrepancies with desktop, so I cleaned those up a bit.

In my journey I found some bugs and deprecations so those were addressed too.

## Preview
Leaning towards this one (plus it has layout fixes):
* [With standard logos](http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12102/25/index.html)

## Comments
Long story short, the site is kind of stuck with old bootstrap, old jquery, and quite a mess of display issues due some hackery. So things don't work quite as they should. I tried updating bootstrap and jquery, but these ended up being incompatible with the site due to said hackery and would need a lot of work.

### CODE REVIEW

**Comment by nswamy** [comment]
played around a bit on the link you provided, 2 observations
1) why do we need on all pages? I think its sufficient to just have it in main page and tutorial pages, everywhere else I feel its a distraction. 
2) Logos are of very poor quality and on my 1080p monitor they look pretty bad, don't they have high def logos?

File: docs/community/contribute.md [hunk 1]
```diff
@@ -62,13 +62,37 @@ To join the MXNet slack channel send request to the contributor mailing list.
 
 ### Social Media
 
-Keep connected with the latest MXNet news and updates on [Twitter](https://twitter.com/apachemxnet) and [Reddit](https://reddit.com/r/mxnet). Also, subscribe to the [MXNet YouTube channel](https://www.youtube.com/channel/UCQua2ZAkbr_Shsgfk1LCy6A).
+Keep connected with the latest MXNet news and updates.
```
Comment (lupesko) [inline docs/community/contribute.md]: This whole section renders bad, with the logos not aligned properly. Needs some work.

Comment (aaronmarkham) [reply docs/community/contribute.md]: I started with the community page and hated how ugly that social section was. That's why I added the nav bar. I'll circle back and fix it.

Also I can add higher resolution images. I might need to requestion a 4k monitor 😋

**Comment by aaronmarkham** [comment]
@swamy for the tutorial pages I had something else in mind. I would provide a different layout.
It would stack like...
Nav
Title
Abstract
Prerequisites
Tags
Share this page buttons
Video if available
Content
CTAs:
* Related tutorials links
* Share this page buttons

With this PR I was testing a simple concept that uses Sphinx's templating system and the site's current layout as a precursor. 

SK's advise was to redo the site to fix all of the bootstrap and Sphinx hacks so it is easy to update and will behave as expected. I won't attempt anything fancy after what I just went through to get even this working. 

I can turn off the include for certain sections. I can add the text menu to the nav to align it with mobile. I can bump the res of the images. After that, it's pretty hard right now. 

Also, I think the buttons should call some attention. We want people to click on them. I'd like to see more votes on limiting which pages they appear.

**Comment by nswamy** [comment]
@aaronmarkham feel free to raise this discussion on the dev@ and seek more opinions. At the moment I feel the design needs a little more work and is not cohesive with the current website layout.

**Comment by aaronmarkham** [comment]
I'm using svg images instead now, so things should look good an whatever resolution, plus they're small. And yes, I put them in the mxnet repo because otherwise they wouldn't work without switching to embedding and tl;dr: no thanks. 

I also posted the [images and associated Sketch files to the web-data repo](https://github.com/dmlc/web-data/tree/master/mxnet/social) in case anyone (like a designer!) wants to tinker with improving the images. 

The contribute page looks better now... and also uses the svg images.

**Comment by aaronmarkham** [comment]
@nswamy I fixed the responsive design for your higher res monitor. Though I'm sure the css could use an overhaul.

File: docs/community/contribute.md (hunk 1 above)
Comment (lupesko) [reply docs/community/contribute.md]: Cool.
> I might need to requestion a 4k monitor 😋

I'm sure you can find a manager at AWS that will approve such a request...

**Comment by lebeg** [comment]
I tested on my Mac / Chrome / 34-inch (3440 x 1440) and the social icons don't look good. 
screen shot 2018-08-15 at 11 46 47
I my opinion the icons are distracting a bit. But the main problem is the complexity of their layout. Do you think it would make sense to try them on the left menu instead? The one that gets hidden on tight mobile sceens.

**Comment by aaronmarkham** [comment]
@lebeg Take a look at the [preview on your mobile](http://34.201.8.176/versions/social_media_update/) or just resize the screen to pretty small - the social links are text in that hamburger menu, but only for "mobile", and that view doesn't trigger until under 600px. So then desktop would never see them. I could just add text to the desktop nav instead of buttons, and the CSS complexity goes away.

The current CSS and implementation of boostrap appears pretty messed up. It behaves inappropriately, and I found that the entanglements are huge. Various JS scripts are injecting or overwriting stuff after boostrap has rendered which could explain why bootstrap doesn't work as expected. Like there's a [footer bug](https://github.com/apache/incubator-mxnet/pull/11921#issuecomment-410153362) that me, Krishnan, or SK couldn't figure out. Maybe someone else can, but it seems like it is basically a start-over job with the design, or hack on what we have. 

I can tinker with the CSS a bit more and try to get rid of that crowding you see - make buttons smaller increase padding and so forth, but I'm going to have to find a better monitor as what I have does get that high of resolution. 

I like incremental improvement. If anyone else wants to take a stab at the same feature, that's cool. Or fix/upgrade the bootstrap implementation. That would be awesome. 

I just think we really should improve the marketing game on the site and try to boost MXNet's social numbers, and I don't think waiting for a site redesign is ideal.

**Comment by simoncorstonoliver** [comment]
I agree that it's probably enough to put it on the landing page only.

Note that we have a friendly URL for Youtube now: https://www.youtube.com/apachemxnet

**Comment by aaronmarkham** [comment]
@simoncorstonoliver Thanks for your feedback. I only include the buttons on the home page now. I've updated the Youtube link.

**Comment by nswamy** [comment]
could you show a variation using the logos provided by the platforms ?

**Comment by aaronmarkham** [comment]
@nswamy This one uses the logos from the platforms rather than trying to conform them to the mxnet blue background.
http://34.201.8.176/versions/social_media_update_v2/

**Comment by lebeg** [comment]
screen shot 2018-08-24 at 17 39 04
Maybe adding some paddings and reducing the size of the buttons might help improving their appearance.

**Comment by aaronmarkham** [comment]
Ok, here's the latest:
http://34.201.8.176/versions/social_media_update_v2/

I tested on a high res monitor and fixed the padding so there shouldn't be overlap.
I had to refactor to allow for the promo.
@nswamy - Do you still request changes?

**Comment by simoncorstonoliver** [comment]
Looking good on my monster monitor

**Comment by aaronmarkham** [comment]
@nswamy - Do you still request changes?

**Comment by lebeg** [comment]
screen shot 2018-08-29 at 10 55 43
 Looks good!

**Comment by nswamy** [comment]
Please check the site on different resolutions, on my mac the logos look small

**Comment by aaronmarkham** [comment]
@lupesko - do you still request changes? The latest is hosted here: http://34.201.8.176/versions/social_media_update_v2/

**Comment by vandanavk** [comment]
@nswamy ping for review/approval

**Comment by aaronmarkham** [comment]
Since ApacheCon is almost over, I should remove that logo, and rebase all of this to make sure it's all working as expected. That would take care of both things, but I don't want to do that if there's not a consensus on this feature.

**Comment by vrakesh** [comment]
@nswamy requesting a review/approval for this PR.

**Comment by Roshrini** [comment]
@aaronmarkham Can you please resolve conflicts so that we can move forward with this PR?

**Comment by aaronmarkham** [comment]
@Roshrini Ok, resolved conflicts... let's see if it clears CI.

**Comment by aaronmarkham** [comment]
Looks fine... Travis just timed out. So this ready for review/approval/merge.

**Comment by ankkhedia** [comment]
@aaronmarkham Were you able to work on the broken images?
If that part is fixed we can add reviewers to go forward.

**Comment by aaronmarkham** [comment]
@ankkhedia Yes, the preview link is updated and the images are all there.

**Comment by ankkhedia** [comment]
@nswamy ping for reviewing again as all the comments has been addressed.

**Comment by kalyc** [comment]
@aaronmarkham thanks for your contribution! I see the CI build is failing - could you re-trigger the build with an empty commit?

**Comment by vandanavk** [comment]
@mxnet-label-bot update [Website, pr-awaiting-testing]

**Comment by aaronmarkham** [comment]
Closing this PR. I can try another round of adding social buttons later. The other updates on the community page are already in #13705.
